File: test-files/sample.py
"""
Sample Python file for testing MCP server
"""

def greet(name: str) -> str:
    """Greet someone by name."""
    return f"Hello, {name}!"


def calculate_sum(a: int, b: int) -> int:
    """Calculate the sum of two numbers."""
    return a + b


class Calculator:
    """A simple calculator class."""
    
    def __init__(self):
        self.result = 0
    
    def add(self, value: int) -> int:
        """Add a value to the result."""
        self.result += value
        return self.result
    
    def subtract(self, value: int) -> int:
        """Subtract a value from the result."""
        self.result -= value
        return self.result
    
    def reset(self) -> None:
        """Reset the calculator."""
        self.result = 0


def main():
    """Main function to test the calculator."""
    calc = Calculator()
    calc.add(10)
    calc.add(5)
    print(f"Result: {calc.result}")
    
    message = greet("World")
    print(message)
    
    total = calculate_sum(10, 20)
    print(f"Sum: {total}")


if __name__ == "__main__":
    main()

File: test-files/sample.ts
/**
 * Sample TypeScript file for testing MCP server
 */

interface Person {
    name: string;
    age: number;
}

function greet(person: Person): string {
    return `Hello, ${person.name}! You are ${person.age} years old.`;
}

class Calculator {
    private result: number = 0;

    add(value: number): number {
        this.result += value;
        return this.result;
    }

    subtract(value: number): number {
        this.result -= value;
        return this.result;
    }

    getResult(): number {
        return this.result;
    }

    reset(): void {
        this.result = 0;
    }
}

function calculateSum(a: number, b: number): number {
    return a + b;
}

function main(): void {
    const person: Person = { name: "Alice", age: 30 };
    console.log(greet(person));

    const calc = new Calculator();
    calc.add(10);
    calc.add(5);
    console.log(`Result: ${calc.getResult()}`);

    const sum = calculateSum(10, 20);
    console.log(`Sum: ${sum}`);
}

main();



